subprojects/gst-plugins-bad/sys/msdk/gstmsdkdec.c

```diff
@@ -1960,23 +1789,21 @@ gst_msdkdec_decide_allocation (GstVideoDecoder * decoder, GstQuery * query)
       goto failed_to_create_pool;
     }
   }
-
-
+#ifndef _WIN32
+  GstAllocator *allocator = NULL;
   if (gst_query_get_n_allocation_params (query) > 0) {
     gst_query_parse_nth_allocation_param (query, 0, &allocator, NULL);
-    if (!(GST_IS_MSDK_VIDEO_ALLOCATOR (allocator) ||
-            GST_IS_MSDK_DMABUF_ALLOCATOR (allocator) ||
-            GST_IS_MSDK_SYSTEM_ALLOCATOR (allocator)))
```

Comment:
The code logic changes here, the GST_IS_MSDK_SYSTEM_ALLOCATOR() is deleted, which means it does not have ds_has_no_msdk_allocator set. Is that the way you want?

Reply:
> The code logic changes here, the GST_IS_MSDK_SYSTEM_ALLOCATOR() is deleted, which means it does not have ds_has_no_msdk_allocator set. Is that the way you want?

Yes, because we no longer use old msdk allocator kinds of stuff and only recognize va/d3d11 allocator.